src/shm_main.c

```diff
@@ -56,9 +56,26 @@ struct _conn_list_entry {
 
 /**
  * @brief Linked list of all active connections in this process.
+ *
+ * Each sysrepo connection maintains a POSIX advisory lock on its lockfile. These
+ * locks allow other sysrepo processes to validate if a tracked connection is
+ * still alive. However a process closing ANY file descriptor to a lockfile on
+ * which it holds an advisory lock results in the lock being immediately
+ * released. To avoid this condition this linked list tracks all open connections
+ * within the current process along with the open file descriptor used to create
+ * the advisory lock. When testing for aliveness (sr_shmmain_conn_lock) this list
+ * is checked first to see if the connection ID is owned by this process. Only
+ * when that check fails will the lock test open (and later close) a file handle
+ * to the lockfile for testing the lock. This list is used by the disconnect logic
+ * to close the filehandle which releases the lock. Programs which do not cleanly
+ * disconnect (eg crash) will have the lock removed automatcially as the
+ * terminated process is cleaned up.
  */
-static conn_list_entry *conn_head = NULL;
-static pthread_mutex_t *conn_list_lock = NULL;
+typedef struct {
+    conn_list_entry *conn_head;
+    pthread_mutex_t *conn_list_lock;
+} sr_locked_list_t;
+sr_locked_list_t local_conn_list = { .conn_head=NULL, .conn_list_lock=NULL };
```

Comment:
This all is probably not needed, so please just use the anonymous struct like this:
```C
static struct {
    conn_list_entry *head;
    pthread_mutex_t *lock;
} conn_list;
```
Not even needed to zero it since it is global, but you can do it in this form, too, if you want.

Reply:
Updated.  Thanks.